### ISSUE TITLE
Feature request: Support for arm-hf (hardware floating point) architecture

### ISSUE THREAD

**Issue by tapirbait** [corbinlc/gnuroot#47]
Hi corbin,

first off, thanks for your work :)  
I would be happy to have a debian image that is armhf in addition to armel, that is already available. I want to run some dsp-related programs, like `sox`.
Probably, i could try to fiddle around by myself. I already have the armel wheezy, which has multiarch support and also, i could try to `debootstrap` (hoping that this works in proot) what i need. It is that i do not know what to do then, if i had a debootstrapped image, how to include it into the list of distros of gnuroot.  
Do you have any ideas? That would help me.

**Comment by corbinlc** [comment]
I will have a major release coming in the not too distant future which will support arm-hf.  In addition, at that point, everything will be open source, including all the build scripts.  So, it will be relatively easy to drop in a different root if someone desires at that point.

**Comment by tapirbait** [comment]
Am happy to hear that. I think an option to extract a custom rootfs tar gz/bz2 from sdcard would suffice as there are lots of prebuilt arm-hf rootfs around.  
Using the armel debian, i could also debootstrap an own armhf rootfs and copy that to sdcard.  
Its just that i don't know how to extract it into app dir and proot into that using gnuroot.

**Comment by tapirbait** [comment]
Hi, I found a sort of workaround for this.  
I could use the existing armel debian to extract an arm hard floating point rootfs to `/data/data/champion.gnuroot/app_install/roots/arch`.
I wanted to give arch linux a try, debian armhf would've worked as well.
I also made a copy of corresponding busybox and proot in `/data/data/champion.gnuroot/app_install/support/arch`. I just used the same files as debian wheezy armel has for that.
Then, an `arch` entry appeared in the list of distros to launch. However the resulting arch `proot` command misses some bindings that debian has, but I don't really need them.  
The explanation why this all works is that the usage of the hardware floating point unit of an arm processor does not need any additional support by proot. Its just a matter of whether the CPU will 'understand' the additional instructions or not. In my device, it does. Memory-wise a floating point number is still a floating point number.
Its similar to running IA32 SSE instructions on a CPU capable of this. The underlying OS, be it windows 95 or whatever, does not need to provide any extra effort as SSE is possible in IA32 protected mode as long as the CPU supports it. Yet it wouldn't work on a 486 running win95.
And what also won't work is running x86_64 code on a system that doesn't support that (e.g. x86 32bit) as the whole memory model is different, pointers are 64bit, etc..
